src/femr/labelers/omop_inpatient_admissions.py

```diff
@@ -109,9 +109,14 @@ def get_outcome_times(self, patient: Patient) -> List[datetime.datetime]:
     def get_prediction_times(self, patient: Patient) -> List[datetime.datetime]:
         """Return end of admission as prediction timm."""
         times: List[datetime.datetime] = []
-        for __, discharge_time in get_inpatient_admission_discharge_times(patient, self.ontology):
+        admission_times = set()
+        for admission_time, discharge_time in get_inpatient_admission_discharge_times(patient, self.ontology):
             prediction_time: datetime.datetime = self.prediction_time_adjustment_func(discharge_time)
+            # Ignore patients who are readmitted the same day they were discharged b/c of data leakage
+            if prediction_time.replace(hour=0, minute=0, second=0, microsecond=0) in admission_times:
```

Comment:
I think the previous logic might be wrong. Just updated with a simple way of checking whether current admission date <= previous discharge date (including "<" just in case there are overlapping visit events).